distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/ServerCli.java

```diff
@@ -107,11 +107,18 @@ public void execute(Terminal terminal, OptionSet options, Environment env, Proce
 
         // we are running in the foreground, so wait for the server to exit
         int exitCode = server.waitFor();
+        checkExit(exitCode);
         if (exitCode != ExitCodes.OK) {
             throw new UserException(exitCode, "Elasticsearch exited unexpectedly");
         }
     }
 
+    /**
+     * A post-exit hook to optionally perform additional processing before the command terminates
+     * @param exitCode the server process exit code
+     */
+    protected void checkExit(int exitCode) {}
```

Comment:
Slight tweak to naming `onExit` - there's no action this class expects overrides to do, this is just a notify method